looprpc/client.proto

```diff
@@ -666,3 +697,124 @@ message LsatToken {
     */
     string storage_name = 8;
 }
+
+message GetLiquidityConfigRequest{}
+
+message LiquidityConfig{
+    /*
+    Set to true if we should include private channels in our balance
+    calculations.
+    */
+    bool include_private = 1;
+
+    /*
+    The target to apply liquidity rules to.
+    */
+    LiquidityTarget target = 2;
+
+    /*
+    A rule to be applied to the target. This value may not be set if setting
+    target to none, and must be supplied otherwise.
+    */
+    LiquidityRule rule = 3;
+}
+
+enum LiquidityTarget{
+    NONE = 0;
+    NODE = 1;
+    PEER = 2;
+    CHANNEL =3;
+}
+
+enum LiquidityRuleType{
+    UNKNOWN = 0;
+    THRESHOLD = 1;
+}
```

Comment:
Necessary?

Reply:
The aim here is to allow a single `LiquidityRule` message, which we expect different fields to be set for depending on the type field that's set. The alternative is a `oneOf`, which I think we want to avoid.

> Choosing between raising questions to rpc users about the meaning of 'unknown' vs reminding users about setting a field, I'd prefer the former.

Can't reply to the other [comment](https://github.com/lightninglabs/loop/pull/265#discussion_r478294590) on this because GH is being strange, but I really feel that we should have the unknown zero value to distinguish between default zero value and intentionally set types. We'd never need to for returning values, only to make sure the user knows what they are setting.